src/test/java/nextstep/subway/utils/RestAssuredTemplate.java

```diff
@@ -0,0 +1,83 @@
+package nextstep.subway.utils;
+
+import io.restassured.RestAssured;
+import io.restassured.response.ExtractableResponse;
+import io.restassured.response.Response;
+import org.springframework.http.MediaType;
+
+import java.util.Map;
+
+public final class RestAssuredTemplate {
+    private final String baseUrl;
+    private final String baseUrlAndId;
+
+    public RestAssuredTemplate(final String baseUrl) {
+        this.baseUrl = baseUrl;
+        this.baseUrlAndId = baseUrl + "/{id}";
+    }
+
+    public ExtractableResponse<Response> get() {
+        return RestAssured.given().log().all()
+                .accept(MediaType.APPLICATION_JSON_VALUE)
+                .when()
+                .get(baseUrl)
+                .then().log().all()
+                .extract();
+    }
+
+    public ExtractableResponse<Response> get(final Long id) {
+        return RestAssured.given().log().all()
+                .accept(MediaType.APPLICATION_JSON_VALUE)
+                .pathParam("id", id)
+                .when()
+                .get(baseUrlAndId)
+                .then().log().all()
+                .extract();
+    }
+
+    public ExtractableResponse<Response> post(final Map<String, String> params) {
+        return RestAssured.given().log().all()
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .body(params)
```

Comment:
여기서 `Map<String, String>`을 전달해야 하는 이유가 있을까요?

실제 코드를 작성해 테스트해보진 않았는데 아래와 같은 형태로도 사용할 수 있지 않을까요?

```suggestion
    public <T> ExtractableResponse<Response> post(final T body) {
        return RestAssured.given().log().all()
                .contentType(MediaType.APPLICATION_JSON_VALUE)
                .body(body)
```

여기 로직 때문에 `BaseRequest` 및 아래와 같은 로직을 추가해주신 것으로 보입니다.

```java
public abstract class BaseRequest<E> {
    private static final ObjectMapper objectMapper = new ObjectMapper();

    protected abstract E toEntity();

    public final Map<String, String> toMap() {
        Map<String, Object> result = objectMapper.convertValue(this, Map.class);

        return result.keySet()
                .stream()
                .collect(Collectors.toMap(key -> key, key -> String.valueOf(result.get(key))));
```

비즈니스 로직에 존재하는 모든 Request가 테스트에서 사용할 toMap 메소드 때문에 BaseRequset를 상속하는 부분은 테스트를 위해 너무 많은 부분을 희생했다는 생각이 들어요.

Reply:
메소드 제네릭을 사용하는 피드백 감사합니다!
DTO에는 상속이 필요없다는 피드백 반영하겠습니다.